spec/std/range_spec.cr

```diff
@@ -79,6 +79,89 @@ describe "Range" do
+    (1..5).union(6..10).should eq(1..10)  # Adjacent ranges
+    (1..5).union(10..15).should eq(1...1) # Disjoint integer ranges
+
+    # Float ranges
+    (1.0..5.5).union(3.2..7.8).should eq(1.0..7.8)  # Overlapping float ranges
+    (1.0..2.5).union(3.0..4.0).should eq(1.0...1.0) # Non-overlapping float ranges
+
+    # String ranges
+    ('a'..'e').union('c'..'g').should eq('a'..'g')  # Overlapping string ranges
+    ('a'..'c').union('e'..'f').should eq('a'...'a') # Non-overlapping string ranges
+    ('a'..'c').union('d'..'f').should eq('a'..'f')  # Adjacent string ranges
+
+    # Time ranges
+    t1 = Time.local(2024, 10, 1)
+    t2 = Time.local(2024, 10, 5)
+    t3 = Time.local(2024, 10, 10)
+    t4 = Time.local(2024, 10, 15)
+    (t1..t2).union(t2..t3).should eq(t1..t3)  # Adjacent time ranges
+    (t1..t2).union(t3..t4).should eq(t1...t1) # Disjoint time ranges
+
+    # Exclusive ranges
+    (1...5).union(5...10).should eq(1...10)           # Adjacent exclusive integer ranges
+    (1.0...5.5).union(3.2...7.8).should eq(1.0...7.8) # Overlapping exclusive float ranges
+    ('a'...'e').union('c'...'g').should eq('a'...'g') # Overlapping exclusive string ranges
+    (t1...t2).union(t2...t3).should eq(t1...t3)       # Adjacent exclusive time ranges
+
+  end
+
+  it "does intersection" do
+    # Int ranges
+    (1..5).intersection(3..7).should eq(3..5)   # Overlapping integer ranges
+    (1..5).intersection(6..10).should eq(1...1) # Non-overlapping integer ranges
+    (1..5).intersection(5..10).should eq(5..5)
+    (1...1).intersection(0..10).should eq(1...1)
+
+    # Float ranges
+    (1.0..5.5).intersection(3.2..7.8).should eq(3.2..5.5)  # Overlapping float ranges
+    (1.0..2.5).intersection(3.0..4.0).should eq(1.0...1.0) # Non-overlapping float ranges
+
+    # String ranges
+    ('a'..'e').intersection('c'..'g').should eq('c'..'e')  # Overlapping string ranges
+    ('a'..'c').intersection('d'..'f').should eq('a'...'a') # Non-overlapping string ranges
+
+    # Time ranges
+    t1 = Time.local(2024, 10, 1)
+    t2 = Time.local(2024, 10, 5)
+    t3 = Time.local(2024, 10, 10)
+    t4 = Time.local(2024, 10, 15)
+    (t1..t3).intersection(t2..t4).should eq(t2..t3)  # Overlapping time ranges
+    (t1..t2).intersection(t3..t4).should eq(t1...t1) # Non-overlapping time ranges
+
+    # Exclusive ranges
+    (1...5).intersection(3...7).should eq(3...5)             # Overlapping exclusive integer ranges
+    (1.0...5.5).intersection(3.2...7.8).should eq(3.2...5.5) # Overlapping exclusive float ranges
+    ('a'...'e').intersection('c'...'g').should eq('c'...'e') # Overlapping exclusive string ranges
+    (t1...t3).intersection(t2...t4).should eq(t2...t3)       # Overlapping exclusive time ranges
+  end
+
+  it "overlaps?" do
+    # Int ranges
+    (1..5).overlaps?(3..7).should eq(true)   # Overlapping integer ranges
+    (1..5).overlaps?(6..10).should eq(false) # Non-overlapping integer ranges
```

Comment:
Another quick check for dependency on order of ranges and added a range with very slight overlap (`1..6` and `6..10`).

> [!WARNING]
> The test for `1..6` and `6..10` fails

```suggestion
    # Int ranges
    (1..5).overlaps?(3..7).should eq(true)   # Overlapping integer ranges
    (3..7).overlaps?(1..5).should eq(true)   # Overlapping integer ranges
    (1..6).overlaps?(6..10).should eq(true)  # Overlapping integer ranges
    (1..5).overlaps?(6..10).should eq(false) # Non-overlapping integer ranges
    (6..10).overlaps?(1..5).should eq(false) # Non-overlapping integer ranges
```